### PR TITLE
[weak-napi] Add type for `weak-napi` package

### PR DESCRIPTION
**Pull request by lh0x00**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by lamhieu-vk** [comment]
This PR to add type for [weak-napi](https://github.com/node-ffi-napi/weak-napi) package.

**Comment by lamhieu-vk** [comment]
@andrewbranch please help me review this PR, many thanks ^^.

File: types/weak-napi/weak-napi-tests.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+const anyVar = null as any;
+
+if (weak.isWeakRef(anyVar)) {
+    const a = anyVar; // WeakRef<any>
+}
+
+if (weak.isDead(weakReference)) {
+    const a = weakReference; // WeakRef<undefined>
```
Comment (andrewbranch) [inline types/weak-napi/weak-napi-tests.ts]: FYI you can use `// $ExpectType WeakRef<undefined>` and it’ll actually be validated as such

**Review by andrewbranch — APPROVED** [review]

File: types/weak-napi/weak-napi-tests.ts (hunk 1 above)
Comment (lamhieu-vk) [reply types/weak-napi/weak-napi-tests.ts]: @andrewbranch I updated it

**Review by andrewbranch — APPROVED** [review]

**Pull request merged**